wait_for_it/wait_for_it.py

```diff
@@ -98,6 +91,13 @@ async def _wait_until_available_and_report(reporter, host, port):
     show_default=True,
     help="Timeout in seconds, 0 for no timeout",
 )
+@click.option(
+    "-s",
+    "--service",
+    metavar="host:port",
+    multiple=True,
+    help="Services to test, in one of the formats 'host:port', '[v6addr]:port' or 'https://...'",
```

Comment:
```suggestion
    help="Services to test, in one of the formats 'host:port', 'ipv4:port', '[ipv6]:port' or 'scheme://host:port'",
```

Reply:
I think `scheme` is better than raw `https` here.

Reply:
Good idea about `ipv4`.

For the URL, some things improved and some things got worse:
- `scheme` mis-implies (at least to me) that more than `http` and `https` are supported but they are not, as of today.
- A URL authinfo, path, query, fragment are all tolerated so are well-formed syntax but are not mentioned. Full URLs work, see below.
- No `scheme://ipv4:port` and not `scheme://[ipvp]:port`?

```console
# wait-for-it -s https://github.com/clarketm/wait-for-it/pull/30/files
[*] Waiting 15 seconds for github.com:443
[+] github.com:443 is available after 0 seconds
```

How do we go from here, what are your key requirements?

Reply:
No hard requirements, just thinking about what makes the most sense. What are your thoughts on using regex syntax – something like this? 
```text
(https?://)(hostname|ipv4|ipv6)(:port)?(/pathname)?(?search)?(#fragment)?
```

Reply:
The regex example misses the square brackets around `ipv6` and square brackets conflict with regex character classes so we'd need to escape them a la `(hostname|ipv4|\[ipv6\])`, same for the query question mark. If we went with that idea, I'd end up with…
```
(https?://)?(user:pass@)?(hostname|ipv4|\[ipv6\])(:port)?(/path)?(\?query)?(#fragment)?
```
…which effectively describes a URL.

Not an easy problem to solve. I think I'm hoping for:
- Something readable
- That explains itself
- That mentions IPv6 square brackets, abstract or concrete
- Briefly mentions URL support

Reply:
PS: If you feel like cherry-picking c2a619c6bfa626bd73f93c7dda3e67f21f4d1644 and doing a release before we find "the  perfect help text" here, please feel free.

Reply:
https://pypi.org/project/wait-for-it/2.1.0/